phpBB/phpbb/template/twig/extension/icon.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\template\twig\extension;
+
+use phpbb\template\twig\environment;
+
+class icon extends \Twig\Extension\AbstractExtension
+{
+	/** @var \phpbb\user */
+	protected $user;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param \phpbb\user	$user		User object
+	 */
+	public function __construct(\phpbb\user $user)
+	{
+		$this->user = $user;
+	}
+
+	/**
+	 * Returns the name of this extension.
+	 *
+	 * @return string The extension name
+	 */
+	public function getName()
+	{
+		return 'icon';
+	}
+
+	/**
+	 * Returns a list of filters to add to the existing list.
+	 *
+	 * @return \Twig\TwigFilter[]
+	 */
+	public function getFilters()
+	{
+		return [
+			new \Twig\TwigFilter('png_path', [$this, 'png_path'], ['needs_environment' => true]),
+		];
+	}
+
+	/**
+	 * Returns a list of functions to add to the existing list.
+	 *
+	 * @return \Twig\TwigFunction[]
+	 */
+	public function getFunctions()
+	{
+		return [
+			new \Twig\TwigFunction('Svg_clean', [$this, 'svg_clean'], ['needs_environment' => true]),
+		];
+	}
+
+	/**
+	 * Create a path to a PNG template icon.
+	 *
+	 * @param environment	$environment	Twig environment object
+	 * @param string		$icon			The icon name
+	 * @return string
+	 */
+	public function png_path(environment $environment, $icon)
+	{
+		$board_url	= defined('PHPBB_USE_BOARD_URL_PATH') && PHPBB_USE_BOARD_URL_PATH;
+		$web_path	= $board_url ? generate_board_url() . '/' : $environment->get_web_root_path();
+		$style_path	= $this->user->style['style_path'];
+
+		return "{$web_path}styles/{$style_path}/template/icons/png/{$icon}.png";
```

Comment:
This should be in theme folder

Reply:
Okay, btw this will not handle 'style inheritance'. Is that correct behaviour?

Reply:
What do you mean?

Reply:
If the icon is not present in the user's style it will display the 'No image'.
It will not inherit from any parent style (eg. Chameleon).